docs/template_plugin/src/template_plugin.cpp

```diff
@@ -61,14 +62,13 @@ std::shared_ptr<ngraph::Function> TransformNetwork(const std::shared_ptr<const n
         new_types.emplace_back(parameter->get_element_type());
     }
 
-    auto clonedNetwork = ngraph::specialize_function(std::const_pointer_cast<ngraph::Function>(function),
-        new_types, new_shapes, std::vector<void *>(new_types.size(), nullptr), constFolding, shareConsts);
-
-    auto transformedNetwork = clonedNetwork;
-    transformedNetwork->set_friendly_name(function->get_friendly_name());
+    auto transformedNetwork = ngraph::clone_function(*function);
 
     // 2. Perform common optimizations and device-specific transformations
     ngraph::pass::Manager passManager;
+    if (constFolding) {
```

Comment:
CF already registered in CommonOptimizations so there is no need to register it here

Reply:
Does it mean that I should remove this `if`?

Reply:
Fixed